packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -100,8 +100,11 @@ class FlutterDevice {
     if (flutterViews == null || flutterViews.isEmpty)
       return;
     for (FlutterView view in flutterViews) {
-      if (view != null && view.uiIsolate != null)
+      if (view != null && view.uiIsolate != null) {
+        // Manage waits specifically below.
```

Comment:
In my change, I copied John's code here: https://github.com/flutter/flutter/commit/0770c3c14fafc2e37632e087b6225f3955294b78#diff-62dbce17da9d1284c7f14ae07e67df33L412 but not the TODO to wait for the exit commands. I don't know why the exit commands are/were not awaited.